client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -93,10 +104,19 @@ const UIComponent = (props: ControlProps) => {
           loading={isFetching}
           optionKey="name"
           onChange={(_, value) => value && onChange(value)}
+          onInputChange={(
+            _event: React.SyntheticEvent<Element, Event>,
+            _value: string,
+            reason: string
+          ) => {
+            if (reason === 'clear') {
+              onChange(null);
+            }
+          }}
           getOptionLabel={option => option.name}
           value={period ? { label: period.name ?? '', ...period } : null}
           isOptionEqualToValue={(option, value) => option.id === value.id}
-          clearable={false}
+          clearable={props.uischema.options?.['clearable'] ?? false}
```

Comment:
I wanted to be generic and not change existing behaviour, so this is an optional input in report ui.

However these components probably shouldn't be so specific I reckon?

@CarlosNZ also had some thoughts on this.